pg_search/src/postgres/utils.rs

```diff
@@ -25,19 +25,21 @@ use chrono::{NaiveDate, NaiveTime};
 use pgrx::itemptr::{item_pointer_get_both, item_pointer_set_all};
 use pgrx::*;
 
-/// Finds and returns the first `USING bm25` index on the specified relation, or [`None`] if there
-/// aren't any
+/// Finds and returns the `USING bm25` index on the specified relation with the
+/// highest OID, or [`None`] if there aren't any.
 pub fn locate_bm25_index(heaprelid: pg_sys::Oid) -> Option<PgRelation> {
     unsafe {
         let heaprel = PgRelation::open(heaprelid);
-        for index in heaprel.indices(pg_sys::AccessShareLock as _) {
-            if !index.rd_indam.is_null()
-                && (*index.rd_indam).ambuild == Some(crate::postgres::build::ambuild)
-            {
-                return Some(index);
-            }
-        }
-        None
+        let indices = heaprel.indices(pg_sys::AccessShareLock as _);
+
+        // Find all bm25 indexes and keep the one with highest OID
```

Comment:
This also needs to filter out any indexes where `rd_indexvalid` is false.

Indexes that are created CONCURRENTLY will have that set to to true while the index is being built.  Behind the scenes, the catalog entry gets committed with that flag set to false and when the build is finished Postgres sets it to true.

Reply:
yeah, it looks right.